concepts/0566-issuer-hosted-custodidal-agents/README.md

```diff
@@ -0,0 +1,108 @@
+
+Hosting agents for users might sound like a costly endeavor, but hosted agents contain an advantage. Most hosted agents will only be used by their owners for a small amount of time, most likely similar to their interaction with the centralized system it replaces. This means that the costs are substantially lower than hosting a full agent.
+
+## Hosted Agent Interaction
+
+Hosted agents have some particular challenges in providing effective user interaction. Detailed below are several options that can be used alone or in combination.
+
+### Browser Based
+Providing a browser based user interface for a user is a common solution when the user will have access to a computer. Authentication will likely use something familiar like a username and password.
+
+### Authorizing Actions
+The user will often need a way to authorize actions that their agent will perform.
+A good option for this is via the use of a basic cell phone through SMS text messages or voice prompts.
+Less urgent actions can use an email sent to the user, prompting the user to login and authorize the actions.
+
+### Offline / Paper based
+At times the user will have no available technology for their use. In this case, providing QR codes printed on paper with accompanying instructions will allow the user to facilitate verifier (and perhaps another issuer) access to their cloud agent.
+QR Codes, such as those detailed in the Out Of Band Protocol, can contain both information for connecting to agent AND an interaction to perform.
+Presenting the QR code for scanning can serve as a form of consent for the prescribed action within the QR code.
+Printed QR codes can be provided by the issuer at the time of custodial agent creation, or from within a web interface available to the user.
+
+### Kiosk based
+Kiosks can be useful to provide onsite interaction with a hosted agent. Kiosk authentication might take place via username and password, smartcard, or USB crypto key, with the possible inclusion of a biometric.
+Kiosks must be careful to fully remove any cached data when a session closes.
+Any biometric data used must be carefully managed between the kiosk and the hosted agent.
+
+### Smartphone App
+While it is common for a smartphone app to be an agent by itself, there are cases where a smartphone app can act as a remote for the hosted agent. In this iteraction, keys, credentials, and other wallet related data is held in the custodial agent. The mobile app acts as a remote viewer and a way for the user to authorize actions taken by the custodial agent.
+
+## Best Practices
+
+The following best practices should be followed to ensure proper operation and continued transition to a fully realized SSI architecture.
+Most of these practices depend upon and support one another.
+
+### Defend the SSI architecture
+When issuers host custodial agents, care must be taken to avoid shortcuts that would violate SSI architecture. Deviations will frequently frequently lead to incompatibilities.
```

Comment:
```suggestion
When issuers host custodial agents, care must be taken to avoid shortcuts that would violate SSI architecture. Deviations will frequently lead to incompatibilities.
```